web-ui/src/main/resources/catalog/views/default/directives/partials/mdactionmenu.html

```diff
@@ -36,17 +36,27 @@
       <i class="fa fa-user"></i>&nbsp;
       <span data-translate="">transferOwnership</span>
     </a></li>
-    <li data-ng-if="user.canEditRecord(md) && user.isReviewerOrMore()">
+    <li data-ng-if="user.canEditRecord(md) && user.isReviewerOrMore()"
+        data-ng-class="(md.isPublished() || (info['metadata.workflow.allowPublishInvalidMd'] === 'true') || 
+          (!md.isPublished() && (info['metadata.workflow.allowPublishInvalidMd'] === 'false') &&
+          (!md.hasValidation() || (md.hasValidation() && md.isValid())))) ? '' : 'disabled'"
+        title="{{(!md.isPublished() ? (md.isValid() ? 'mdvalid' : (!md.hasValidation() ? 'mdnovalidation': 'mdinvalid')) : '') | translate }}"
+        >
       <a data-ng-click="mdService.publish(md)">
         <i class="fa"
            data-ng-class="md.isPublished() ? 'fa-lock' : 'fa-unlock'"></i>&nbsp;
+        <i class="fa" data-ng-if="!md.isPublished() && md.hasValidation()"
+           data-ng-class="md.isValid() ? 'fa-check' : 'fa-close fa-minus-circle'"></i>
+        <i class="fa fa-question" data-ng-if="!md.isPublished() && !md.hasValidation()"></i>
```

Comment:
Not a UI expert, but maybe moving the validation status icon to the right side would preserve alignment of menus and I would use the same icons as in the editor validation panel with green and red background maybe ?